File: TeddyPet_BE/src/main/java/fpt/teddypet/application/dto/response/shop/TimeSlotResponse.java
package fpt.teddypet.application.dto.response.shop;

import com.fasterxml.jackson.annotation.JsonProperty;
import fpt.teddypet.domain.enums.scheduling.DayTypeEnum;
import fpt.teddypet.domain.enums.scheduling.SlotTypeEnum;

import java.time.LocalTime;

public record TimeSlotResponse(
        Long id,
        Long serviceId,
        DayTypeEnum dayType,
        LocalTime startTime,
        LocalTime endTime,
        Integer maxCapacity,
        Integer currentBookings,
        SlotTypeEnum slotType,
        String notes,
        String status,
        @JsonProperty("version")
        Long version
) {
}

File: teddypet_mobile/lib/data/models/response/service/time_slot_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'time_slot_response.g.dart';

@JsonSerializable()
class TimeSlotResponse {
  final int id;
  final int serviceId;
  final String startTime;
  final String endTime;
  final String? dayType;
  final String? status;
  final int? currentBookings;
  final int? maxCapacity;
  final int? version;

  TimeSlotResponse({
    required this.id,
    required this.serviceId,
    required this.startTime,
    required this.endTime,
    this.dayType,
    this.status,
    this.currentBookings,
    this.maxCapacity,
    this.version,
  });

  factory TimeSlotResponse.fromJson(Map<String, dynamic> json) =>
      _$TimeSlotResponseFromJson(json);

  Map<String, dynamic> toJson() => _$TimeSlotResponseToJson(this);
}
